### DISCUSSION TITLE
How long is the runner registration token valid for?

### DISCUSSION THREAD

**Discussion by toast-gear** [actions/runner#1799] [Q&A] [answered]
👋  question in the title, upon a runner registration token being generated how long is it valid for before it is rejected?

Ref https://github.com/actions-runner-controller/actions-runner-controller/issues/1295

is this still right? https://github.community/t/api-to-generate-runners-token/16963/8#M4012, an hour

**Comment by toast-gear** [comment]
https://github.com/actions/runner/issues/1339#issuecomment-920316886 I assume this is still valid

**Comment by rmg** [comment, marked as answer]
Per the API docs for generating registration tokens, it's valid for one hour.
- [enterprise](https://docs.github.com/en/rest/actions/self-hosted-runners#create-a-registration-token-for-an-enterprise) `/enterprises/{enterprise}/actions/runners/registration-token`
- [organization](https://docs.github.com/en/rest/actions/self-hosted-runners#create-a-registration-token-for-an-organization) `/orgs/{org}/actions/runners/registration-token`
- [repository](https://docs.github.com/en/rest/actions/self-hosted-runners#create-a-registration-token-for-a-repository) `/repos/{owner}/{repo}/actions/runners/registration-token`

**Comment by interifter** [comment]
What is interesting is, at least for me, a google search has this higher up than the actual documentation @rmg pointed us to.
